src_c/draw.c

```diff
@@ -115,6 +115,13 @@ aaline(PyObject *self, PyObject *arg, PyObject *kwargs)
         return NULL; /* Exception already set. */
     }
 
+    if (!blend) {
+        PyErr_WarnEx(PyExc_DeprecationWarning,
```

Comment:
Missing check of the return value from `PyErr_WarnEx`  If it returns `-1` then an exception was thrown ( see https://docs.python.org/3/c-api/exceptions.html#issuing-warnings), which in turn requires an early return from this function.